tracer/build/_build/PrepareRelease/GenerateIntegrationDefinitions.cs

```diff
@@ -25,70 +30,32 @@ public static List<InstrumentedAssembly> GetAllIntegrations(ICollection<string>
                 var assemblyLoadContext = new CustomAssemblyLoadContext(Path.GetDirectoryName(path));
                 var assembly = assemblyLoadContext.LoadFromAssemblyPath(path);
 
-                callTargetIntegrations = callTargetIntegrations.Concat(GetCallTargetIntegrations(assembly));
+                callTargetIntegrations = callTargetIntegrations.Concat(GetCallTargetIntegrations(assembly, dependabotJsonFile));
 
                 assemblyLoadContext.Unload();
             }
 
             return callTargetIntegrations.ToList();
         }
 
-        static IEnumerable<InstrumentedAssembly> GetCallTargetIntegrations(Assembly assembly)
+        static IEnumerable<InstrumentedAssembly> GetCallTargetIntegrations(Assembly assembly, AbsolutePath dependabotJsonFile)
         {
-            var definitionsClass = assembly.GetType("Datadog.Trace.ClrProfiler.InstrumentationDefinitions");
-            var definitionsMethod = definitionsClass
-               .GetMethod("GetAllDefinitionsNative", BindingFlags.Static | BindingFlags.NonPublic);
-            var derivedDefinitionsMethod = definitionsClass
-                   .GetMethod("GetAllDerivedDefinitionsNative", BindingFlags.Static | BindingFlags.NonPublic);
-            var getIntegrationIdMethod = definitionsClass
-                   .GetMethod("GetIntegrationId", BindingFlags.Static | BindingFlags.NonPublic);
-            var getAdoNetIntegrationIdMethod = definitionsClass
-                   .GetMethod("GetAdoNetIntegrationId", BindingFlags.Static | BindingFlags.Public);
-
-            var integrationIdExtensionsClass = assembly.GetType("Datadog.Trace.Configuration.IntegrationIdExtensions");
-            var toStringFastMethod = integrationIdExtensionsClass.GetMethod("ToStringFast", BindingFlags.Static | BindingFlags.Public);
-
-            var structDefinition = assembly.GetType("Datadog.Trace.ClrProfiler.NativeCallTargetDefinition");
-
-            Array definitions = (Array)definitionsMethod.Invoke(null, Array.Empty<object>());
-            Array derivedDefinitions = (Array)derivedDefinitionsMethod.Invoke(null, Array.Empty<object>());
-
+            var definitions = JsonConvert.DeserializeObject<CallTargetDefinitionSource[]>(File.ReadAllText(dependabotJsonFile));
```

Comment:
You're probably better off using System.Text.Json here seeing as that's what you used to serialize them...

```suggestion
            var options = new JsonSerializerOptions();
            options.IncludeFields = true;
            using var fs = File.OpenRead(dependabotJsonFile);
            var definitions = System.Text.Json.JsonSerializer.Deserialize<CallTargetDefinitionSource[]>(fs, options);
```